src/GitlabCli/MergeRequests.psm1

```diff
@@ -80,10 +85,14 @@ function Get-GitlabMergeRequest {
         $Query['wip'] = $IsDraft ? 'yes' : 'no'
     }
     
-    return Invoke-GitlabApi GET $Path $Query -MaxPages $MaxPages -WhatIf:$WhatIf | 
+    $result = Invoke-GitlabApi GET $Path $Query -MaxPages $MaxPages -WhatIf:$WhatIf | 
         ForEach-Object {
             $_ | New-WrapperObject -DisplayType 'Gitlab.MergeRequest'
         }
+
+    if($Branch -ne $null) {
+        $result | Where-Object SourceBranch -eq $Branch
```

Comment:
rather than filtering here, you could add a `source_branch` to the API request

Reply:
Yeah this was a shortcut. After making the change the query is much faster resolving too!